lib/utils.pm

```diff
@@ -154,11 +154,14 @@ sub wait_boot {
         assert_screen 'displaymanager', 200;
         wait_idle;
         if (get_var('DM_NEEDS_USERNAME')) {
-            type_string $username;
+            type_string "$username\n";
         }
```

Comment:
and you're certain that DESKTOP=gnome and DM_NEEDS_USERNAME will never appear in the same situation? Because normally I would expect an elsif here